metrics/apis/metrics/v1alpha1/types.go

```diff
@@ -0,0 +1,56 @@
+// Copyright 2016 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/api/v1"
+)
+
+// resource usage metrics of a node.
+type NodeMetrics struct {
+	unversioned.TypeMeta `json:",inline"`
+	v1.ObjectMeta        `json:"metadata,omitempty"`
+
+	// The following fields define time interval from which metrics were
+	// collected in the following format [Timestamp-Window, Timestamp].
+	Timestamp unversioned.Time     `json:"timestamp"`
+	Window    unversioned.Duration `json:"window"`
+
+	// The memory usage is the memory working set.
```

Comment:
This is one of possibility although we agree in proposal (https://github.com/kubernetes/kubernetes/pull/25279) to use just ResourceMemory constant. We may revisit this decision later since this is just alpha version of the API.